docs/generator.py

```diff
@@ -253,10 +283,170 @@ def generate_doc_files(self, dest_dir):
     def _generate_from_template(self, dest_dir, filename):
         template = self._jinja_env.get_template(filename)
         content = template.render(**self._template_ctx)
-        output_filename = filename[:-len(self.J2_SUFFIX)]
+        output_filename = self._get_file_name_from_template_name(filename)
         self._write_generated_file(dest_dir, output_filename, content)
 
 
+class ResourceModelDocGenerator(ModelDocGenerator):
+
+    """Slightly customised ModelDoc generation class. Required for generation of FTD API Documentation"""
+
+    MODEL_TEMPLATE = 'resource_model.md.j2'
+
+    @staticmethod
+    def _get_index_data(index_name, index_list):
+        return {'index_list': index_list}
+
+
+class ResourceDocGenerator(BaseDocGenerator, OperationDocGenerationMixin):
+    """Generates documentation for the HTTP resources defined in the
+    Swagger specification. Describes available endpoints with their methods,
+    models, query/body/path parameters, etc.
+    Documentation is written using Markdown markup language.
+    """
+
+    OPERATION_TEMPLATE = 'resource_operation.md.j2'
+    OVERVIEW_TEMPLATE = 'resource_overview.md.j2'
+    CONFIG_TEMPLATE = 'resource_config.json.j2'
```

Comment:
Can we reuse `resource_config.json.j2` to generate tag's `config.json`? Both `resource_config.json.j2` and `config.json.j2` define simple structure with a bunch of `.md` files and optional index page.

Reply:
done